packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowFields.tsx

```diff
@@ -47,9 +47,19 @@ export class DetailsRowFields extends React.PureComponent<IDetailsRowFieldsProps
               ? onRender(item, itemIndex, column)
               : getCellText(item, column);
 
+          // generate a key that auto-dirties when content changes, to force the container to re-render, to trigger animation
+          let key: number | string;
```

Comment:
I strongly recommend that you provide a way for the `key` to be provided to the `DetailsRow` directly via some pathway, so apps can tailor the value to suit their needs. Most of the time, the cell content is a JSX element, which might simply be a generic class component and not the value.